Add pep match to dbNSFP plugin

helensch opened the pull request:
PR for #382
For further details and examples see: ENSVAR-3930

aparton reviewed (commented):
Everything works as expected, but I have a couple of UX questions before merging:

- Should we be turning on the pep_match functionality by default?
- Certain parameters only work when they're given in a specific order. For example, consequence needs to be first, then the file name, then pep match. This is fine and functionality that we have in many of our plugins, however if we're allowing users to supply the parameter name then this implies that the order isn't necessary as the assigning of values to each parameter is done by the user. We're treating required and optional parameters differently, and they're also mixed up in order. While this isn't something that's been explicitly caused by this change, it has added to it and it might be worth fixing it before the plugin becomes too muddled.

helensch commented:
@aparton thank you for the review. 

I have updated the error message.

As the correct match is that used by pep_match=1 it was agreed that pep_match should be on by default.

I have added a ticket for the issues raised by specification of options ENSVAR-4042.  By default the pep_match option does not need to specified so the change in most cases should not add to the option issues flagged.

aparton commented:
merged into release/104 and master
